src/main/java/minegame159/meteorclient/systems/modules/misc/AutoWither.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * This file is part of the Meteor Client distribution (https://github.com/MeteorDevelopment/meteor-client/).
+ * Copyright (c) 2021 Meteor Development.
+ */
+
+package minegame159.meteorclient.systems.modules.misc;
+
+import meteordevelopment.meteorclient.events.world.TickEvent;
+import meteordevelopment.meteorclient.settings.BoolSetting;
+import meteordevelopment.meteorclient.settings.Setting;
+import meteordevelopment.meteorclient.settings.SettingGroup;
+import meteordevelopment.meteorclient.systems.modules.Categories;
+import meteordevelopment.meteorclient.systems.modules.Module;
+import meteordevelopment.meteorclient.utils.player.FindItemResult;
+import meteordevelopment.meteorclient.utils.player.InvUtils;
+import meteordevelopment.meteorclient.utils.player.PlayerUtils;
+import meteordevelopment.meteorclient.utils.world.BlockUtils;
+import meteordevelopment.orbit.EventHandler;
+import net.minecraft.block.Blocks;
+import net.minecraft.item.Items;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+
+public class AutoWither extends Module {
+
+    private SettingGroup sgGeneral = settings.getDefaultGroup();
+
+    private Setting<Boolean> rotate = sgGeneral.add(new BoolSetting.Builder()
+            .name("rotate")
+            .description("Whether or not to rotate while building")
+            .defaultValue(true)
+            .build()
+    );
+
+    // Part 1 : Build wither directly infront of player (complete)
+    // Part 2 : Build wither based on where the player is looking
+    public AutoWither() {
+        super(Categories.World, "auto-wither", "Automatically builds withers.");
+    }
+
+    @EventHandler
+    private void onTick(TickEvent.Post event) {
+
+        // Check for soulsand and skull in hotbar
+        if (!hasEnoughMaterials()) {
+            error("(default)Not enough resources in hotbar");
+            toggle();
+            return;
+        }
+
+        // Find direction of player
+        // North, South, East, West
+        Direction dir = getDirection(mc.gameRenderer.getCamera().getYaw() % 360);
+
+
+        // Aligns player to center of block to avoid obstructing wither placement
+        PlayerUtils.centerPlayer();
+
+
+        // Check if we can place a wither
+        BlockPos blockPos = mc.player.getBlockPos();
+        blockPos = blockPos.offset(dir);
+
+        if (!isValidSpawn(blockPos, dir)) {
+            error("(default)Unable to spawn wither, obstructed by non air blocks");
+            toggle();
+            return;
+        }
+
+
+        // Build the wither
+        info("(default)Spawning wither");
+        spawnWither(blockPos, dir);
+        toggle();
```

Comment:
Would be better if this wasn't a single use thing, most auto withers just place a sequence of withers in the direction you are facing (like you would if you were griefing).